src/main/java/com/google/sps/servlets/AllMarkersServlet.java

```diff
@@ -43,17 +43,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
   
   private Collection<Marker> getMarkers() {
-    /* Collection to hold all markers from datastore */
-    Collection <Marker> allMarkers = new ArrayList<>();
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Marker");
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      double lat = (double) entity.getProperty("lat");
-      double lng = (double) entity.getProperty("lng");
-      Marker marker = new Marker(lat, lng);
-      allMarkers.add(marker);
-    }
+     /* Collection to hold all markers from datastore */
+    Collection<Marker> allMarkers = new ArrayList<>();
+     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+     Query query = new Query("Recommendation");
```

Comment:
Since we will only be needing the lat and lng, [projection (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyProjection)(com.google.appengine.api.datastore.PropertyProjection) might be able to make the interaction with datstore lighter and reduce the memory needed to store the result
Using projection we can get only the necessary attribute, instead of all the attributes.

Reply:
Thanks for the link! This sounds like a great idea.

Reply:
As you have already read up on the documentation and have implemented this in the algorithm section, would you be able to commit the code / suggest a change? With the timeframes, I think this might be helpful.